src/Payum/Paypal/ExpressCheckout/Nvp/Action/CancelAction.php

```diff
@@ -41,9 +37,12 @@ public function execute($request)
      */
     public function supports($request)
     {
-        return
-            $request instanceof Cancel &&
-            $request->getModel() instanceof \ArrayAccess
-        ;
+        if (false == ($request instanceof Cancel && $request->getModel() instanceof \ArrayAccess)) {
+            return false;
+        }
+
+        // it must take into account only common payments, recurring payments must be cancelled by another action.
+        $model = $request->getModel();
+        return isset($model['TRANSACTIONID']);
```

Comment:
> btw could you please move this class out of Api folder. to a parent one while you are on it https://github.com/Payum/Payum/blob/master/src/Payum/Paypal/ExpressCheckout/Nvp/Action/Api/CancelRecurringPaymentsProfileAction.php

I was hoping to discuss to re-factor this action a little bit, currently it does a replace on the model but the [ManageRecurringPaymentsProfileStatus response](https://developer.paypal.com/docs/classic/api/merchant/ManageRecurringPaymentsProfileStatus_API_Operation_NVP/#managerecurringpaymentsprofilestatus-response-fields) returns just the PROFILEID. It does not makes any sense since the PROFILEID won't change, however all the previous data in the model will be lost.

I think that better approach is to implement very similar 2 gateway calls like in [transaction Cancel action](https://github.com/Payum/Payum/blob/master/src/Payum/Paypal/ExpressCheckout/Nvp/Action/CancelAction.php#L35), what do you think?